webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/types/InferringGravsearchTypeInspector.scala

```diff
@@ -793,14 +836,149 @@ class InferringGravsearchTypeInspector(nextInspector: Option[GravsearchTypeInspe
     }
 
     /**
-      * Runs all the inference rules repeatedly until no new type information can be found.
-      *
-      * @param iterationNumber    the current iteration number.
-      * @param intermediateResult the current intermediate result.
-      * @param entityInfo         information about Knora ontology entities mentioned in the Gravsearch query.
-      * @param usageIndex         an index of entity usage in the query.
-      * @return a new intermediate result.
-      */
+     * Gets the iri of the type information.
+     *
+     * @param typeInfo    a GravsearchEntityTypeInfo.
+     * @return the IRI of the typeInfo as a [[SmartIri]].
+     **/
+    def iriOfGravsearchTypeInfo(typeInfo: GravsearchEntityTypeInfo): SmartIri = {
+        typeInfo match {
+            case propertyTypeInfo: PropertyTypeInfo => propertyTypeInfo.objectTypeIri
+            case nonPropertyTypeInfo: NonPropertyTypeInfo => nonPropertyTypeInfo.typeIri
+            case _ => throw GravsearchException(s"There is an invalid type")
+        }
+    }
+
+    /**
+     * Sanitize determined results. If there were multiple resource types, replace with common base class
+     *
+     * @param lastResults this type inspection results.
+     * @param querySchema the ontology schema that the query is written in.
+     **/
+    def sanitizeInconsistentResourceTypes(lastResults: IntermediateTypeInspectionResult, querySchema: ApiV2Schema, entityInfo: EntityInfoGetResponseV2): IntermediateTypeInspectionResult = {
+
+        //This method gets the isResourceType flag of GravsearchEntityTypeInfo
+        def getIsResourceFlags(typeInfo: GravsearchEntityTypeInfo): Boolean = {
+            typeInfo match {
+                case propertyTypeInfo: PropertyTypeInfo => propertyTypeInfo.objectIsResourceType
+                case nonPropertyTypeInfo: NonPropertyTypeInfo=> nonPropertyTypeInfo.isResourceType
+                case _ => throw GravsearchException(s"There is an invalid type")
+            }
+        }
+
+        // This method finds the common base class.
+        def findCommonBaseClass(typesToBeChecked: Set[GravsearchEntityTypeInfo]): SmartIri = {
```

Comment:
What if there is no common base class? For example, what if `typesToBeChecked` contains `beol:letter` and `knora-api:TextValue`?

Reply:
well, if there is no common base class, it returns knora-base:Resource.
findCommonBaseClass is only used if all typesToBeChecked are resource types.